watertap/unit_models/nanofiltration_DSPMDE_0D.py

```diff
@@ -1807,10 +1947,10 @@ def calculate_scaling_factors(self):
         for (t, x, y), v in self.electric_potential.items():
             if iscale.get_scaling_factor(v) is None:
                 iscale.set_scaling_factor(v, 1e4)
-
-        for (t, x), v in self.electric_potential_grad_feed_interface.items():
-            if iscale.get_scaling_factor(v) is None:
-                iscale.set_scaling_factor(v, 1e-2)
+        if hasattr(self, "electric_potential_grad_feed_interface"):
+            for v in self.electric_potential_grad_feed_interface.values():
+                if iscale.get_scaling_factor(v) is None:
+                    iscale.set_scaling_factor(v, 1e-2)
```

Comment:
Is this going to be sufficient for the scaling of this variable? Do we know anything about the range of values for the gradients?